src/core/raster/qgscolorrampshader.h

```diff
@@ -130,7 +130,7 @@ class CORE_EXPORT QgsColorRampShader : public QgsRasterShaderFunction
     /** Maximum size of the color cache. The color cache could eat a ton of
      * memory if you have 32-bit data
      * @deprecated will be removed in QGIS 3.0 */
```

Comment:
I don't think we deprecate private members. There is not much use for it since we have the full control over every usage anyway so instead of a two-step process (1. deprecate 2. remove) like for public members on which 3rd party code depends we can just remove it right away.

In such cases I normally just add a `// TODO QGIS 3: remove this` note.

FYI, you can also use

    Q_NOWARN_DEPRECATED_PUSH
    useDeprecatedMethodOrVariable
    Q_NOWARN_DEPRECATED_POP

which comes in handy when tests are written that checks if deprecated code still works until removed.